### DISCUSSION TITLE
Help with knowing when a function call is made in the main thread using OpenAI

### DISCUSSION THREAD

**Discussion by olixva** [spring-projects/spring-ai#1559] [Q&A]
Hi everyone,

I’m currently working with an OpenAI chat client that includes function-calling capabilities. The problem I’m facing is that when the function is called, I need to have a way to know in my main thread (where I make the call) that the function has been triggered. This is important because I need to return some necessary data to my front end once the function is executed.

Could anyone suggest how I can achieve this? How can I be notified or track in my main thread that the function has been called successfully?

Thanks in advance for your help!

**Comment by che-make** [comment]
+1

**Comment by PPKRex** [comment]
+1
